src/utils/Traverse.js

```diff
@@ -2,12 +2,13 @@ import React from 'react';
 import WrapInMarks from './WrapInMarks';
 import { v4 as uuid } from 'uuid';
 import ApmTableOfContents from '../atoms/ApmTableOfContents/ApmTableOfContents';
+import ApmTableOfContentsClassic from '../atoms/ApmTableOfContents/ApmTableOfContentsClassic';
 
 const Traverse = (props) => {
-  if (
-    props.nodeData?.content &&
-    props.nodeData?.content[0]?.type === 'apm_table_of_contents'
-  ) {
+  const toc = props.nodeData?.content.find(
+    (node) => node.type === 'apm_table_of_contents'
+  );
+  if (toc?.attrs?.anchors) {
```

Comment:
Could you just use the regular WrapInMarks behavior below and not worry about rending the rest of the nodes in the new TOC handler?

Reply:
Yes  but then I have to  move the logic out of Traverse and back to the Table of Contents component

Reply:
Maybe that's  not a big deal though will try it

Reply:
So this works well for the new style TOC  but  will not work  for the legacy style TOC  because you have to send all the nodes with it to be able to fish out the headings.   We could make a choice to not support old style ones.  They  don't work right now  anyways on node sites.